### ISSUE TITLE
Prep for V2

### ISSUE THREAD

**Issue by BrandonShar** [inertiajs/inertia-django#51]
Hi folks! I'm beginning the process of squaring Inertia Django up to get ready for the Inertia V2 improvements that we're going to need, but it's now been several years since I've worked on a Django project professionally. If there's anyone in the community who is using Inertia Django actively and would be interested in collaborating with me on the V2 upgrade, please comment here or reach out!

**Comment by eddyojb88** [comment]
I want to leave a comment to reflect my sense of excitement about a v2! I'm not an appropriate fit as I have just started using inertia and also found it a struggle to set it up, resorting to django-breeze to smooth over the process. Hoping this will be addressed in v2!

**Comment by sopelj** [comment]
Hi. I am relatively new to Inertia (I only have one small project using it), but I do have lots of experience with Django. If there's anything I can do to help, I'd be happy to.

**Comment by BrandonShar** [comment]
@eddyojb88 thank you that's great feedback! I also maintain Inertia Rails and we've felt that same onboarding pain over there. We got luckily in Rails and a top notch company helped build an installer, but that doesn't mean we can't improve the Django situation! Was there anything in particular that tripped you up or was it just too confusing in general?

@sopelj Thanks for the reply! Lots of experience with Django is exactly what would be most helpful. If you could potentially help me test out alpha/beta releases and sanity check Django standards, that would be amazing. I'm not a native python developer, so some of my Django code may not be as expected 😄

**Comment by pmdevita** [comment]
I use Django professionally and the Inertia integration in a few hobby projects, I'd be willing to help out!

**Comment by eddyojb88** [comment]
@BrandonShar

Some thoughts:

-  "[At its core, Inertia is essentially a client-side routing library](https://inertiajs.com/how-it-works)"

-  "[Inertia has no client-side routing](https://inertiajs.com/), nor does it require an API. Simply build controllers and page views like you've always done! Inertia works great with any backend framework, but it's fine-tuned for [Laravel](https://laravel.com/)." 

-  I find the interaction between the Django server, Inertia, Vite and React confusing. I have struggled for hours to understand what is processing and serving what, which didn't help when debugging when I went to setup. I am having to try it out during my evenings so probably a bit tired and dumb but in terms of helping the unitiated, perhaps a diagram would help here

3) I think an opinionated hello-world quick setup is super valuable to on-ramp a new user and let them break it to learn quickly (something like a Django-Breeze)

**Comment by eddyojb88** [comment]
A diagram like [this](https://www.advantch.com/blog/django-react-and-inertiajs/) helps

**Comment by sopelj** [comment]
> @sopelj Thanks for the reply! Lots of experience with Django is exactly what would be most helpful. If you could potentially help me test out alpha/beta releases and sanity check Django standards, that would be amazing. I'm not a native python developer, so some of my Django code may not be as expected 😄

Definitely, I'd love to. Just let me know if you have some branches to check or if you want me to check the repo in general. If it's in general, I could open some issues or maybe start some discussions for suggestions (which I'd be happy to help implement if desired).

**Comment by sopelj** [comment]
I agree that `django-breeze` does help one get started. It might be nice, although, I don't like how it includes views you can't override. It would be nice if it had class based views that one could easily extend and add themselves. Especially for form views. It would be awesome to have something built-in that one could simply extend. Something like was mentioned in #30 

It is also not super intuitive to integrate Django's URL patterns with Inertia. It would be good if we had an example with that too. Especially for localized URLs. I have not yet found a good way of implementing that. Something like [django-js-reverse](https://github.com/vintasoftware/django-js-reverse), but I don't think that handles localization.

**Comment by keinstn** [comment]
Hi, I've just started using Inertia and Django, but I have a lot of experience on Python. I'd be happy if I could help you something!

**Comment by svengt** [comment]
Hi all, I opened a pull request #55 to implement the history encryption feature of inertia v2. Feedback is welcome!

**Comment by mrfolksy** [comment]
Hey all, I've also just started using inertia, and wanted to use the deferred prop feature in 2.0, I opened a PR [#56](https://github.com/inertiajs/inertia-django/pull/56) to implement this. 

Feedback most welcome, especially as this is my first contribution to this project (as well as any other project for quite a few years)!  I'm also happy to help on other areas! I have a reasonable amount of experience with Django and python.

**Comment by BrandonShar** [comment]
@svengt and @mrfolksy thanks for the PRs!! Both are a huge help to getting us to V2!

**Comment by BrandonShar** [comment]
Closing in favor of keeping V2 updates in #57 

I'd love to keep the conversation going about better onboarding though so I'll open another issue for that.
